tools/codegen/codespec/api_to_provider_spec_mapper.go

```diff
@@ -2,69 +2,218 @@
 package codespec
 
 import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	high "github.com/pb33f/libopenapi/datamodel/high/v3"
+	low "github.com/pb33f/libopenapi/datamodel/low/v3"
+
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	"github.com/mongodb/terraform-provider-mongodbatlas/tools/codegen/config"
 	"github.com/mongodb/terraform-provider-mongodbatlas/tools/codegen/openapi"
 )
 
-// using blank identifiers for now, will be removed in follow-up PRs once logic for conversion is added
-func ToProviderSpecModel(atlasAdminAPISpecFilePath, configPath string, resourceName *string) *Model {
-	_, err := openapi.ParseAtlasAdminAPI(atlasAdminAPISpecFilePath)
+func ToProviderSpecModel(atlasAdminAPISpecFilePath, configPath string, resourceName string) (*Model, error) {
+	apiSpec, err := openapi.ParseAtlasAdminAPI(atlasAdminAPISpecFilePath)
 	if err != nil {
 		panic(err)
```

Comment:
Do we need to panic here? Looks like we are doing it in `main.go` anyway?

Reply:
lmc

Reply:
panic is not bad for a tool, but it's a bit "ugly" as you'll get the execution stack trace and a lot of info that can confuse the tool user. 

In case we want to exit the program, a nicer alternative can be to use log.Fatal or log.Fatalf that shows a message and do os.Exit(1),  so it'll be flagged as failure for scripts or GHAs

Reply:
updated to use log.Fatalf wherever applicable